src/PIL/ImImagePlugin.py

```diff
@@ -341,6 +341,8 @@ def _save(im: Image.Image, fp: IO[bytes], filename: str) -> None:
         # or: SyntaxError("not an IM file")
         # 8 characters are used for "Name: " and "\r\n"
         # Keep just the filename, ditch the potentially overlong path
+        if isinstance(filename, bytes):
+            filename = filename.decode("ascii")
```

Comment:
Is this also fixing a bug?

Reply:
In main at the moment, save handlers can receive strings or bytes as the filename.
https://github.com/python-pillow/Pillow/blob/53e82e49c548420cc84a1ce233096f5b06d2cf37/src/PIL/Image.py#L2454
https://github.com/python-pillow/Pillow/blob/53e82e49c548420cc84a1ce233096f5b06d2cf37/src/PIL/Image.py#L231-L232

Part of this PR is propagating that out to some plugins that were only using strings.